updateHostsWindows.bat

```diff
@@ -1,61 +1,61 @@
-:: Admin privileges script based on https://stackoverflow.com/a/10052222
-::
-::
-@ECHO OFF
-TITLE Update Hosts
-
-:: Check if we are administrator. If not, exit immediately.
-:: BatchGotAdmin
-:-------------------------------------
-REM  --> Check for permissions
-    IF "%PROCESSOR_ARCHITECTURE%" EQU "amd64" (
->nul 2>&1 "%SYSTEMROOT%\SysWOW64\cacls.exe" "%SYSTEMROOT%\SysWOW64\config\system"
-) ELSE (
->nul 2>&1 "%SYSTEMROOT%\system32\cacls.exe" "%SYSTEMROOT%\system32\config\system"
-)
-
-REM --> If error flag set, we do not have admin.
-if '%errorlevel%' NEQ '0' (
-    echo Requesting administrative privileges...
-    goto UACPrompt
-) else ( goto gotAdmin )
-
-:UACPrompt
-    echo Set UAC = CreateObject^("Shell.Application"^) > "%temp%\getadmin.vbs"
-    set params= %*
-    echo UAC.ShellExecute "cmd.exe", "/c ""%~s0"" %params:"=""%", "", "runas", 1 >> "%temp%\getadmin.vbs"
-
-    "%temp%\getadmin.vbs"
-    del "%temp%\getadmin.vbs"
-    exit /B
-
-:gotAdmin
-    pushd "%CD%"
-    CD /D "%~dp0"
-
-:BackupHosts
-	:: Backup default hosts file
-	if not exist "%WINDIR%\System32\drivers\etc\hosts.skel" (
-		COPY %WINDIR%\System32\drivers\etc\hosts %WINDIR%\System32\drivers\etc\hosts.skel
-	)
-
-:UpdateHosts
-	:: Update hosts file
-	python updateHostsFile.py --auto --minimise
-	
-	:: Move new hosts file in-place
-	COPY hosts %WINDIR%\System32\drivers\etc\
-
-	:: Flush the DNS cache
-	ipconfig /flushdns
-	
-	:: Summary note
-	pause
+::
+:: This script will first create a backup of the original or the current hosts
+:: file and save it in a file named "hosts.skel".
+::
+:: If the "hosts.skel" file exists, the new hosts file with the customized unified
+:: hosts will be copied to the proper path. Next, the DNS cache will be refreshed.
+::
+:: THIS BAT FILE MUST BE LAUNCHED WITH ADMINISTRATOR PRIVILEGES
+:: Admin privileges script based on https://stackoverflow.com/a/10052222
+::
+
+@echo off
+title Update Hosts
+
+:: Check if we are an administrator. If not, exit immediately.
+:: BatchGotAdmin
+:: Check for permissions
+if "%PROCESSOR_ARCHITECTURE%" equ "amd64" (
+    >nul 2>&1 "%SYSTEMROOT%\SysWOW64\cacls.exe" "%SYSTEMROOT%\SysWOW64\config\system"
+) else (
+    >nul 2>&1 "%SYSTEMROOT%\system32\cacls.exe" "%SYSTEMROOT%\system32\config\system"
+)
+
+:: If the error flag set, we do not have admin rights.
+if %ERRORLEVEL% neq 0 (
+    echo Requesting administrative privileges...
+    goto UACPrompt
+) else (
+    goto gotAdmin
+)
+
+:UACPrompt
+echo Set UAC = CreateObject^("Shell.Application"^) > "%TEMP%\getadmin.vbs"
+set params= %*
+echo UAC.ShellExecute "cmd.exe", "/c ""%~s0"" %params:"=""%", "", "runas", 1 >> "%TEMP%\getadmin.vbs"
+
+"%TEMP%\getadmin.vbs"
+del "%TEMP%\getadmin.vbs"
+exit /b
+
+:gotAdmin
+cd /d "%~dp0"
+
+:BackupHosts
+:: Backup the default hosts file
+if not exist "%WINDIR%\System32\drivers\etc\hosts.skel" (
+    copy /y /v "%WINDIR%\System32\drivers\etc\hosts" "%WINDIR%\System32\drivers\etc\hosts.skel"
```

Comment:
The file does not exist. So `/y` is not needed.

Reply:
We could probably just rename the file instead of copying it, then.

Reply:
If something went wrong with the updater, you will then break the hosts. That also means that you lose protection for at the runtime of the script.

Reply:
`ren` would fail if the file exists anyway. Not important either way since I removed the overwrite flag.

Reply:
Uh ... We are probably talking about the same thing but just to clarify a bit my sentence and what I meant.

You stated:
> We could probably just rename the file instead of copying it, then.

My answer:
> If something went wrong with the updater, you will then lose the hosts. That also means that you lose protection for and at the runtime of the script.

What did I mean? Well, as stated in the comment of the script, it's a backup. 

This means that if we rename `hosts` to `hosts.skel`, for the time of the execution of `updateHostsFile.py`, there will be no protection for the end-user.
So, if something went wrong in the execution of `updateHostsFile.py` you're adding an unexpected behavior (and some work too) for the end-users.